src/Evolution/Systems/CurvedScalarWave/Worldtube/Tags.cpp

```diff
@@ -18,13 +18,41 @@
 #include "Utilities/Gsl.hpp"
 
 namespace CurvedScalarWave::Worldtube::Tags {
-template <size_t Dim>
-void CenteredFaceCoordinatesCompute<Dim>::function(
-    const gsl::not_null<std::optional<tnsr::I<DataVector, Dim, Frame::Grid>>*>
+template <size_t Dim, typename Frame, bool Centered>
+void FaceCoordinatesCompute<Dim, Frame, Centered>::function(
+    const gsl::not_null<std::optional<tnsr::I<DataVector, Dim, Frame>>*> result,
+    const ::ExcisionSphere<Dim>& excision_sphere, const Element<Dim>& element,
+    const tnsr::I<DataVector, Dim, Frame>& coords, const Mesh<Dim>& mesh) {
+  const auto direction = excision_sphere.abutting_direction(element.id());
+  if (direction.has_value()) {
+    const size_t grid_size =
+        mesh.slice_away(direction->dimension()).number_of_grid_points();
+    if (result->has_value()) {
+      destructive_resize_components(make_not_null(&(result->value())),
+                                    grid_size);
+    } else {
+      result->emplace(grid_size);
+    }
+    evolution::dg::project_tensor_to_boundary(make_not_null(&(result->value())),
+                                              coords, mesh, direction.value());
+    if constexpr (Centered) {
+      for (size_t i = 0; i < Dim; ++i) {
+        result->value().get(i) -= excision_sphere.center().get(i);
```

Comment:
This only works for grid coords, so add an `ERROR` in an `if constexpr (Frame is not Grid)` branch.